time service test class added

giba86 opened the pull request:
Time service added, hope Git indicates only 1 file change

giba86 committed:
time service test class added

starlinginferno reviewed (approved):
green ticks aww yea

csadesz commented on src/test/java/rueppellii/backend2/tribes/gameUtility/timeService/TimeServiceTest.java:
```diff
@@ -0,0 +1,118 @@
+
+        buildingLevel = 2;
+        Long TownHallLevelOneAndBuildingLevelTwo = calculationSystemTime +
+                (BUILDING_PROGRESSION_TIME * buildingLevel) / townhallLevel;
```
calculationSystemTime +   (BUILDING_PROGRESSION_TIME * buildingLevel) / townhallLevel

this line could be separated in a different method, because you use it so many times. That method might have 2 input parameters, and you can add it like: buildingTime(1,1); so you dont have to add new value to your variables

giba86 replied:
I modified the class and from now I cal timeService methods directly. Rearranged the variables and put a few in BeforeEach setUp function, however, I feel that those are not necessary there. I change the content of the class via github editor

giba86 committed:
Refactored the class

I changed assertions to JUNIT 5 methods, employed time service functions and cleaned up the variables. In calculateTimeDifferenceOfTroopCreationAtBarrackLevelOneTest function I had to deduct -1 mill sec from the time difference because the run time and rounding rule make a change in the value. Time is up assertions required a progression model in order to calculate the boolean variable. To get true I reduced the progression time; as building and troop progression time is equal I picked and use building progression time. Elpase time and remainer time function aren't tested yet.

csadesz commented on src/test/java/rueppellii/backend2/tribes/gameUtility/timeService/TimeServiceTest.java:
```diff
@@ -0,0 +1,111 @@
+    }
+
+    @Test
+    void calculateTimeOfBuildingCreationTest() {
```
Try to follow this naming convention:
[the name of the tested feature][expected input / tested state][expected behavior]

like

registerNewUserAccount_ExistingEmailAddressGiven_ShouldThrowException()
